src/main/controllers/WorkAddressController.ts

```diff
@@ -53,6 +52,7 @@ export default class WorkAddressController {
         const respondentIndex = getRespondentIndex(req);
         updateWorkAddress(req.session.userCase, req.session.userCase.respondents[respondentIndex]);
       }
+      handleUpdateDraftCase(req, logger);
```

Comment:
I think this should stay inside the no errors block but just after updateWorkAddress

Reply:
If the bug is caused by workAddress not existing on a case there is another scenario to consider. If the claimant says they did not work for the respondent they will not reach this page. Check the conditions on respondentAddress controller